toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```diff
@@ -81,54 +121,117 @@ func (b *BootCustomizer) AddKernelCommandLine(extraCommandLine []string) error {
 	return nil
 }
 
-// Gets the image's configured SELinux mode.
-func (b *BootCustomizer) getSELinuxModeFromGrub() (imagecustomizerapi.SELinuxMode, error) {
+// getSELinuxModeFromGrub gets the image's configured SELinux mode from grub configuration.
+// Returns (mode, found, error) where:
+//   - mode: The detected SELinux mode (or SELinuxModeDefault if not found)
+//   - found: true if grub config exists and was successfully parsed
+//   - error: Any error encountered during parsing
+func (b *BootCustomizer) getSELinuxModeFromGrub() (imagecustomizerapi.SELinuxMode, bool, error) {
```

Comment:
Rename `getSELinuxModeFromGrub` to getSELinuxModeFromCmdline` and then handle the UKI case in this function. This will simplify and de-duplicate the code (e.g. the call to `getSELinuxModeFromConfigFile`).